scripts/backup_with_rotation.sh

```
@@ -0,0 +1,43 @@
+#Last Modified Time: 16/10/2024
+
+if [[ $# -eq 0 ]] || [[ ! -d "$1" ]];
+then
+	echo "Please provide valid directory path which you want to take backup:"
+	exit
+fi
+
+source_dir="$1"
+
+function create_backup {
+	local timestamp=$(date '+%Y-%m-%d_%H-%M-%S')
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)